server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/FilterAllocationDeciderTests.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster.routing.allocation.decider;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ESAllocationTestCase;
+import org.elasticsearch.cluster.EmptyClusterInfoService;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.routing.RoutingTable;
+import org.elasticsearch.cluster.routing.allocation.AllocationService;
+import org.elasticsearch.cluster.routing.allocation.allocator.BalancedShardsAllocator;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.snapshots.EmptySnapshotsInfoService;
+import org.elasticsearch.test.gateway.TestGatewayAllocator;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.crate.testing.Asserts.assertThat;
+import static org.elasticsearch.cluster.routing.ShardRoutingState.INITIALIZING;
+
+
+public class FilterAllocationDeciderTests extends ESAllocationTestCase {
+
+    @Test
+    public void test_allocation_require_cluster_setting() {
+        var clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        var settings = Settings.builder().put("cluster.routing.allocation.require.storage", "hot").build();
+
+        var clusterState = buildClusterState(settings(Version.CURRENT));
+        var allocationService = buildAllocationService(settings, clusterSettings);
+        var routing =  allocationService.reroute(clusterState, "reroute").routingTable();
+
+        var t1 = routing.index("t1");
+        assertThat(t1.shards().size()).isEqualTo(3);
+
+        var shardRouting = t1.shard(0).shards().get(0);
```

Comment:
minor: here and in all other tests, you could add a for loop to assert the 3 shards.

Reply:
I tried that, however it does not improve readability. I think it's easier to read like this.

Reply:
Totally fine to leave as is, it's a personal preference of mine.